### DISCUSSION TITLE
Increasing the SPI_FREQUENCY does not work

### DISCUSSION THREAD

**Discussion by ountheas** [Bodmer/TFT_eSPI#3014] [Q&A - General]
I am using [ILI9488-GT911](https://tr.aliexpress.com/item/1005004083478309.html) display with [NodeMCU-32s](https://imgur.com/iDdEUoN). I have no problems with the display or touch. However, no matter how much I change the SPI_FREQUENCY value, it has no effect.

`Serial.println(spiClockDivToFrequency(spiGetClockDiv(SPI.bus())));`

When I run the command, I see the value **80000000**. However, the refresh rate does not change even if I increase the SPI_FREQUENCY value.

**Setup20_ILI9488.h**
`// See SetupX_Template.h for all options available
#define USER_SETUP_ID 20

#define ILI9488_DRIVER

#define TFT_CS 5
#define TFT_RST 17
#define TFT_DC 16
#define TFT_MOSI 23
#define TFT_SCK 18

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  80000000`

**Comment by Bodmer** [comment]
That approach will not work because the library does not use the default SPI instance, it creates a new spi instance.
Use getSPIinstance:
https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_eSPI.h#L824

**Reply by ountheas** [reply]
Thank you for your reply. I see the same value when I test the way you mentioned.

No matter what value I change SPI_FREQUENCY to, the refresh rate does not increase. However, if I decrease this value, the refresh rate decreases. Values above 15-20 mhz have no effect. Is this due to the hardware limitations of the display?

I also have one more question. My current pins are as follows:

#define TFT_CS 15
#define TFT_RST 17
#define TFT_DC 16
#define TFT_MOSI 23
#define TFT_SCLK 18

Is there a problem here?

Thanks again for your help.

**Reply by Bodmer** [reply]
The SPI frequency can only be set to an integer division of 80MHz, i.e 80MHz,  40MHz, 26.66MHz, 20MHz etc. If you try to set a value between these it will be set to the next nearest lower value. This is a hardware design limitation.
